Update on forcing config

dumontgoulart opened the pull request:
Added options for runs not to have wind, and for temp and pet in Wflow to accept the same data as precip (in case zarr files are not available as it is the case for Freddy). All solutions were thought in a dynamic way to remove hard coded parts

dumontgoulart committed:
first

dumontgoulart committed:
Update execution script and configuration files for workflow improvements

- Adjusted ROOT path in run_snakemake_all_workflow_h7_mzb_kenneth.sh
- Updated config_general_mzb_Kenneth.yml to specify wind forcing and coastal boundary conditions
- Renamed data catalog entries in datacatalog_general___linux.yml from v2 to v3
- Added initial test.py file with hydromt import

dumontgoulart committed:
Update workflow scripts and configuration for improved model run parameters and wind forcing handling

dumontgoulart committed:
Update workflow scripts and configuration for improved model run parameters and coastal forcing handling

dumontgoulart committed:
Update forcing configuration and add execution script for workflow setup

dumontgoulart committed:
Update execution script and configuration for improved workflow settings and forcing parameters

couasnonanais commented on Workflows/00_execution_script_examples/run_snakemake_all_workflow_h7_mzb_freddy.sh:
```diff
@@ -0,0 +1,39 @@
+#SBATCH --exclusive 
+#SBATCH --ntasks=1                  # Number of tasks (analyses) to run
+
+export HDF5_USE_FILE_LOCKING=FALSE
```
Did you see that this resolved the problem with the wflow being stuck? It would be nice to add this line to the other mozambique case as well

dumontgoulart replied:
We used this for the `OSError: [Errno -101] NetCDF: HDF error: '/p/11210221-gtsm-reanalysis/GTSM-ERA5-E_dataset/waterlevel/reanalysis_waterlevel_hourly_1950_01_v2.nc'` regarding GTSM

But for wflow I don't think this helps too much as I still had at least 2 issues with wflow hanging.

couasnonanais commented on Workflows/01_config_snakemake/config_general_MZB.yml:
```diff
@@ -6,38 +6,53 @@ dir_models:  02_Models              # Folder where the base models will be saved
+
+        # Models selections
+        use_dfm: True                                       # use the coastal hydrodynamic model D-FM
+        use_wflow: True   
```
I think right now 'use_wflow' and 'use_wave' are not pointing to really being used? Or does the user get an error if use_wflow is False? Maybe we could for now comment those lines and write in the description that this option has not been added yet

couasnonanais commented on Workflows/01_config_snakemake/config_general_MZB.yml:
```diff
@@ -6,38 +6,53 @@ dir_models:  02_Models              # Folder where the base models will be saved
+        # Models selections
+        use_dfm: True                                       # use the coastal hydrodynamic model D-FM
+        use_wflow: True   
+        use_waves: True                                  # use the hydrological model Wflow      
```
The description does not match the line (it talsk about wflow for waves)

couasnonanais commented on Workflows/01_config_snakemake/config_general_mzb_Kenneth.yml:
```diff
@@ -0,0 +1,44 @@
+        start_time: 20190425 000000                         # Start time of the SFINCS model run in format: YYYYMMDD HHMMSS           
+        end_time: 20190430 000000                           # End time of the SFINCS model run in format: YYYYMMDD HHMMSS          
+        forcing: era5_hourly_zarr                           # forcing dataset for the rainfall data (matching to the DataCatalog handle)
+        wind_forcing: no_wind                          # wind and pressure dataset (used for coastal surge modelling)
```
I understand that here only a set of selected keywords are possible if no specific wind datasets is used. (the SKIP_WIND_KEYWORDS ). Can you add this desciption in the comments of this line?

dumontgoulart replied:
ty, fixed it

dumontgoulart committed:
Update config_general_MZB.yml

dumontgoulart committed:
Update config_general_mzb_Kenneth.yml

dumontgoulart committed:
Refactor forcing parameters in Snakemake configurations and scripts

- Updated configuration files to replace 'forcing' with 'precip_forcing' and 'meteo_forcing' for clarity and consistency across various model setups (config_general_MZB.yml, config_general_MZB3.yml, config_general_honduras.yml, config_general_mzb_Freddy.yml, config_general_mzb_Kenneth.yml, config_general_somerset.yml).
- Modified Snakemake rules to utilize 'meteo_forcing' and 'precip_forcing' instead of 'forcing' in snakefiles (snakefile_CF_data.smk, snakefile_all_wflow_sfincs.smk, snakefile_dfm.smk, snakefile_dfm_cluster.smk, snakefile_sfincs_update.smk, snakefile_wflow.smk).
- Adjusted Python scripts to reflect changes in parameter names for forcing datasets (update_sfincs_coastal_forcing.py, update_forcing_wflow_event.py).

dumontgoulart committed:
final updates - precip separation seems to work fine
